inc/namespace.php

```diff
@@ -598,3 +611,34 @@ function on_aws_guzzle_request_stats( TransferStats $stats ) {
 	}
 	send_trace_to_daemon( $trace );
 }
+
+function redact_metadata( $metadata ) {
+
+	$redact_keys_default = [];
+	if ( function_exists( 'apply_filters' ) ) {
+		$redact_keys_default = apply_filters( 'aws_xray.redact_metadata_keys', $redact_keys_default );
```

Comment:
Rather than doing this with a declarative system "Give me the meta keys, and I'll handle redacting" also _atleast_ add a mechanism to make it function. I.e. pass `$metadata` through a filter and allow the function to strip what it sees fit. 

I'm also not sure that such a declarative approach is a good one -- do we really want to strip all `pwd` POST keys? I think _sure_ it's probably ok, but I'd imagine things hooking in would want to do this a lot more specific "null out this value on these types of requests specifically".

Reply:
Initially I had the option to filter the metadata directly: https://github.com/humanmade/aws-xray/pull/34/commits/6548eaa8e485397fcbdf79d8c7a066db0b5267fe

Is this more in line with what you're looking for?

Reply:
Yup exactly

Reply:
> do we really want to strip all pwd POST keys?

Absolutely. We don't want to be storing these passwords in traces in plaintext. I can't think of a time where we _would_ want to keep such sensitive information in plaintext?

Reply:
That's assuming `pwd` _is_ a password on all requests; which my comment was coming form the point of pointing out that might not be the case. If it was called _password_ then maybe, but even then, I think there are situations, such as `post password` where you probably would log that information (they are plaintext in the DB, and if you had reports of failing password attempts you may want to cross reference the submitted data).

Reply:
In this case, I'm wanting to protect pwd on wp-login.php, at which point we never want to store that. I hear your point that the same key might be re-used on other kinds of requests. I think the aws_xray.redact_metadata filter which I just added below might provide an escape hatch for those instances. Would you agree? I'd want to err on always blocking and conditionally allowing, rather than always allowing and conditionally blocking sensitive information.

Reply:
Yeah I think it's ok to leave it, I was pointing out that I don't think it's certain. `pwd` is a weirdly named key, and we know exactly which pages that can be triggered on, so looking for `wp-login.php` / reset password would also be ok to do, but yeah I think this is fine too. The main point here was to address the general functional filter which we've now got 👍

Reply:
👍 👍